components/market/ticket/TicketPurchaseModal.tsx

```diff
@@ -83,6 +90,77 @@ function TicketPurchaseModal({ open, setOpen, ticket, event }: P) {
     }
   }
 
+  const handlePrimaryTicketPurchase = (
+    contract: any,
+    address: `0x${string}`,
+    price: ethers.BigNumber
+  ) => {
+    contract.methods
+      .buyTicket(
+        `${process.env.NEXT_PUBLIC_WARDEN_API_URL}/ticket/metadata?path=${event._id}/${ticket.ticketMetadata[0].name}`,
+        TicketTypeKey[ticket.ticketType]
+      )
+      .send({ from: address, value: price })
+      .then(async (result: any) => {
+        const ticketId = result.events?.TicketCreated?.returnValues?._ticketId
```

Comment:
add if(result.events?.TicketCreated?.returnValues?._ticketId) to be safe

Reply:
Added check
```ts
if (!result || !result.events || !result.events?.TicketCreated) {
          showErrorAlert({
            type: AlertType.ERROR,
            title: "Ticket purchase failed",
            description: "Smart contract purchas eoperation failed"
          })
          return
        }
```